Make plugins configuration syntax an array

ehmicky opened the pull request:
Fixes #341.

This changes the configuration syntax from:

```yml
plugins:
  pluginId:
    enabled: true
    config:
      foo: bar
```

To:

```yml
plugins:
  - id: pluginId
    enabled: true
    config:
      foo: bar
```

DavidWells commented:
Wow fast work! ⚡️

Just to verify `id` is not required right?

`id` should be an optional field that plugins add if:

1. They need to load multiple copies of themselves
2. If they have outputs that other things need to reference

DavidWells commented:
If `id` is not required this looks good to go.

We will need to update examples here:

- This repo
- https://github.com/DavidWells/jamstack-sf
- @philhawksworth examples
- @sw-yx 's examples
- @sdras's examples & post https://www.netlify.com/blog/2019/10/16/creating-and-using-your-first-netlify-build-plugin/
- and the current invited beta user list https://docs.google.com/spreadsheets/d/1irAPELGq-FmM7N4cNx7I5eLi9vRIMjKKYHpqkexQBEk/edit?usp=sharing

ehmicky commented:
I fixed the `id`, which is now optional. I have added a test for this.

I have also fixed all the examples in the repository.

I have created the following issue for updating the documentation outside of the repository: #351.

Munter commented:
I've updated [checklinks](https://www.npmjs.com/package/netlify-plugin-checklinks), [subfont](https://www.npmjs.com/package/netlify-plugin-subfont) and [hashfiles](https://www.npmjs.com/package/netlify-plugin-hashfiles) to the new config syntax.

Big thumbs up for the error messages when things aren't configured right 👍
